src/com/palmergames/bukkit/TownyChat/listener/EssentialsDiscordHookListener.java

```diff
@@ -0,0 +1,38 @@
+package com.palmergames.bukkit.TownyChat.listener;
+
+import com.palmergames.bukkit.TownyChat.Chat;
+import com.palmergames.bukkit.TownyChat.channels.Channel;
+import com.palmergames.bukkit.TownyChat.channels.channelTypes;
+import net.essentialsx.api.v2.events.discord.DiscordChatMessageEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class EssentialsDiscordHookListener implements Listener {
+  private Chat plugin;
+
+  public EssentialsDiscordHookListener(Chat plugin) {
+    this.plugin = plugin;
+  }
+
+  @EventHandler(ignoreCancelled = true)
+  public void onDiscordChat(DiscordChatMessageEvent event) {
+    Channel channel = plugin.getChannelsHandler().getChannel(event.getPlayer(), plugin.getTownyPlayerListener().directedChat.get(event.getPlayer()));
+    if (channel != null) {
+      event.setCancelled(!channel.isHooked());
```

Comment:
Basing this off of whether a channel is hooked or not will break all of the other plugins using TownyChat, including DeluxeChat. A hooked channel means some other plugin is modifying the channel format.

Reply:
To elaborate: Many servers use DeluxeChat to alter their channels, including the global/general channel that everyone sees.

If we accepted this PR all of those established servers would find their general channel's chat cancelled.

I still am of the opinion that EssentialsDiscord is going to have to choose to be more deterministic on what it chooses to broadcast. We'll see what the other team members have to contribute.